introspection-engine/introspection-engine-tests/tests/tables/mod.rs

```diff
@@ -742,3 +742,119 @@ async fn casing_should_not_lead_to_mix_ups(api: &TestApi) -> TestResult {
 
     Ok(())
 }
+
+#[test_connector(tags(Mysql), exclude(Mariadb))]
+async fn unique_and_index_on_same_field_works_mysql(api: &TestApi) -> TestResult {
+    api.barrel()
+        .execute(|migration| {
+            migration.inject_custom(
+                "create table users (
+                       id serial primary key not null
+                     );",
```

Comment:
I'm less clear on that one, does it mean all `SERIAL PRIMARY KEY` columns on MySQL will suddenly have an `@unique`? Is it likely to cause hard to understand drift in migrate?

Reply:
Since serial is shorthand for Integer Autoincrement Unique this is correct. There are really two separate constraints in the db.